blivet/osinstall.py

```diff
@@ -239,11 +239,11 @@ def _findExistingInstallations(devicetree):
             device.format.mount(options=options, mountpoint=getSysroot())
         except Exception: # pylint: disable=broad-except
             log_exception_info(log.warning, "mount of %s as %s failed", [device.name, device.format.type])
-            device.format.umount(mountpoint=getSysroot())
```

Comment:
Yeah it's little complicated. When you read PR https://github.com/rhinstaller/blivet/pull/223 which I fixed badly with the commit. You can't use `teardown()` because of it won't umount protected disks here. So when you have stage2 or repo HDD connected it's mounted forever and it's blocking repository to load properly. 
So or so after `_findExistingInstallations()` method finish `teardownAll()` should be called. Some potential problem could be the `teardown_all` parameter in the `findExistingInstallations()` method could be set false but I didn't found out calling where this happening so this could be patched for master as follow-up but to f23 this is a good patch though.

Also as @dwlehman wrote:
> it potentially eliminates some unnecessary teardown/setup cycles